src/chrome/chromeConnection.ts

```diff
@@ -100,23 +104,29 @@ export class ChromeConnection {
     /**
      * Attach the websocket to the first available tab in the chrome instance with the given remote debugging port number.
      */
-    public attach(address = '127.0.0.1', port = 9222, targetUrl?: string, timeout?: number): Promise<void> {
-        return this._attach(address, port, targetUrl, timeout)
+    public attach(address = '127.0.0.1', port = 9222, targetUrl?: string, timeout?: number, extraCRDPChannelPort?: number): Promise<void> {
+        return this._attach(address, port, targetUrl, timeout, extraCRDPChannelPort)
             .then(() => { });
     }
 
-    public attachToWebsocketUrl(wsUrl: string): void {
+    public attachToWebsocketUrl(wsUrl: string, extraCRDPChannelPort?: number): void {
         this._socket = new LoggingSocket(wsUrl);
-        this._client = new Client(<WebSocket>this._socket as any);
+        if (extraCRDPChannelPort) {
+            this._crdpSocketMultiplexor = new CRDPMultiplexor(this._socket);
+            new WebSocketToLikeSocketProxy(extraCRDPChannelPort, this._crdpSocketMultiplexor.addChannel('extraCRDPEndpoint')).start();
```

Comment:
Make sure this is shut down cleanly when debugging is done, even when running in debug server mode.